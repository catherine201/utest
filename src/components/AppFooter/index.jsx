import React from 'react';
import { Icon } from 'antd';

export default class AppFooter extends React.Component {
  render() {
    return (
      <div className="text-center">
        <Icon type="cloud" />
        Make by &nbsp;
        <span>{/* <Tag color="blue">EUEN</Tag> */}</span>
      </div>
    );
  }
}
