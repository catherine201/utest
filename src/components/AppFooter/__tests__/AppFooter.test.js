import React from 'react';
// import renderer from 'react-test-renderer';  // 这个不支持antdesign
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AppFooter from '../index';

Enzyme.configure({ adapter: new Adapter() });
describe('(Component) AppHeader -- snapshot', () => {
  it('capture snapshot of AppHeader', () => {
    const footer = shallow(<AppFooter />);
    expect(footer).toMatchSnapshot();
  });
});
