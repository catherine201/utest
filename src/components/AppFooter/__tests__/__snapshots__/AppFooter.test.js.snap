// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AppHeader -- snapshot capture snapshot of AppHeader 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          type="cloud"
        />,
        "Make by  ",
        <span>
          <Tag
            closable={false}
            color="blue"
            prefixCls="ant-tag"
          >
            EUEN
          </Tag>
        </span>,
      ],
      "className": "text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": "cloud",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "Make by  ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Tag
            closable={false}
            color="blue"
            prefixCls="ant-tag"
          >
            EUEN
          </Tag>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "EUEN",
            "closable": false,
            "color": "blue",
            "prefixCls": "ant-tag",
          },
          "ref": null,
          "rendered": "EUEN",
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            type="cloud"
          />,
          "Make by  ",
          <span>
            <Tag
              closable={false}
              color="blue"
              prefixCls="ant-tag"
            >
              EUEN
            </Tag>
          </span>,
        ],
        "className": "text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "cloud",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "Make by  ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Tag
              closable={false}
              color="blue"
              prefixCls="ant-tag"
            >
              EUEN
            </Tag>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "EUEN",
              "closable": false,
              "color": "blue",
              "prefixCls": "ant-tag",
            },
            "ref": null,
            "rendered": "EUEN",
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
